tools/yul_phaser/Chromosome.h

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace solidity::phaser
+{
+
+class Chromosome
+{
+public:
+	Chromosome():
+		m_optimizationSteps{} {}
+	explicit Chromosome(std::vector<std::string> const& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	explicit Chromosome(std::vector<std::string>&& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
```

Comment:
```suggestion
		m_optimizationSteps(std::move(_optimizationSteps)) {}
```

Reply:
Done. I have also replaced these two constructors with one that takes steps by value since that still allows vector to be moved.